canonical-web-and-design/maas-ui
# refactor(ui): store errors with id and action

## Done

- Associate machine errors with the machine and the action that performed the action.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine list and select a few machines and perform an action that returns an error (e.g. open the tag form and create a tag with a symbol in the name), the error should be displayed.
- Go to a machine details page and again perform an action that returns an error, the error should be displayed.

## Fixes

Fixes: canonical-web-and-design/maas-ui#1842.
